core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -271,5 +378,51 @@ private static MetadataUpdate readAddSortOrder(JsonNode node) {
     UnboundSortOrder sortOrder = SortOrderParser.fromJson(sortOrderNode);
     return new MetadataUpdate.AddSortOrder(sortOrder);
   }
+
+  private static MetadataUpdate readSetDefaultSortOrder(JsonNode node) {
+    int sortOrderId = JsonUtil.getInt(SORT_ORDER_ID, node);
+    return new MetadataUpdate.SetDefaultSortOrder(sortOrderId);
+  }
+
+  private static MetadataUpdate readAddSnapshot(JsonNode node) {
+    Snapshot snapshot = SnapshotParser.fromJson(null, node);
+    return new MetadataUpdate.AddSnapshot(snapshot);
+  }
+
+  private static MetadataUpdate readRemoveSnapshots(JsonNode node) {
+    Set<Long> snapshotIds = JsonUtil.getLongSetOrNull(SNAPSHOT_IDS, node);
```

Comment:
When could this be null? I think this is correct (so this comment is minor) but I'd normally expect to use `getLogSet` that handles the null case internally. This is fine because there's a good error message.

Reply:
You're correct. This should never be null (as evidence by the null check below).

I added the `getLongSetOrNull` function essentially because it mirrored the `getIntegerSetOrNull` function and still needed a preconditions check here for the size (given we don't currently support multiple values).

I can update to `getLongSet` given it's added in this PR, but the Precondition check will still be needed.